managed-ledger/src/main/java/org/apache/bookkeeper/mledger/Entry.java

```diff
@@ -48,7 +50,18 @@
     Position getPosition();
 
     /**
-     * Release the resources allocated for this entry
+     * Increments ref-count of entry data
+     */
+    void retain();
+    
+    /**
+     * Recycles entry instance 
+     */
+    void recycle();
```

Comment:
we follow the same semantic (same as ByteBuf -> release() will recycle if ref-count=0). However, we provided this method to support usecase where one wants to use EntryImpl temp to just pass the data(byteBuf) and not disturbing life-cycle of that `data` and recycle that EntryImpl immediately. eg. [OpAddEntry](https://github.com/yahoo/pulsar/pull/318/files/81e154d1597389f60ca3f1678449d408cb32131e#diff-772b4979830bd173bc8fa5a1d52a7588) or [Consumer](https://github.com/rdhabalia/pulsar/blob/81e154d1597389f60ca3f1678449d408cb32131e/pulsar-broker/src/main/java/com/yahoo/pulsar/broker/service/Consumer.java#L201)